PhotonSieveDevelopment.js

```diff
@@ -24,92 +24,96 @@ class Sieve {
         this.totalPinholes = totalPinholes
         this.totalArea = totalArea
         this.fNumber = fNumber
+        this.rings = []
+    }
+
+    addRing(ring) {
+        this.rings.push(ring);
+        this.totalArea = this.totalArea + ring.openArea;
+        this.totalPinholes = this.totalPinholes + ring.pinholes.length;
+        let sieveEquivalentAperture = Math.sqrt(this.totalArea / Math.PI) * 2;
+        this.fNumber = FD / sieveEquivalentAperture;
     }
 }
 
 class Ring {
-    constructor(zone, centerlineRadius, ringWidth, numOfPinholes, openArea){
+    constructor(zone, centerlineRadius){
         this.zone = zone
         this.centerlineRadius = centerlineRadius
-        this.ringWidth = ringWidth
-        this.numOfPinholes = numOfPinholes
-        this.openArea = openArea
+        this.openArea = 0
+        this.pinholes = []
     } 
+
+    addPinhole(pinhole) {
+        this.pinholes.push(pinhole);
+        this.openArea = this.openArea + pinhole.pinholeArea;
+    }
+
+    // conflict checking function can safely ignore pinholes placed on rings other than current 
+    getConflicts(conflictRadius, conflictAzimuth) {
+        let conflictArray = [];
+        let candidateX = conflictRadius * Math.cos(conflictAzimuth);
+        let candidateY = conflictRadius * Math.sin(conflictAzimuth);
+    
+        // loop through all pinholes on current ring, check if min spacing constraint is violated
+
+        for(let i = 0; i < this.pinholes.length; i ++){
+            let conflictDistanceBetweenCenters = distanceBetween(new Point(candidateX, candidateY), this.pinholes[i].point);
+            let conflictDistanceBetweenPerimeters = conflictDistanceBetweenCenters - this.pinholes[i].radius - conflictRadius;
+            if(conflictDistanceBetweenPerimeters <= SPACE_MIN){
+                conflictArray.push(this.pinholes[i]);
+            }
+        }
+
+        return conflictArray
+    }
 }
 
-/*class Point {
+class Point {
     constructor(x,y) {
         this.x = x
         this.y = y
     }
-}*/
+}
 
 class Pinhole {
-    constructor(radius, azimuth, pinholeArea) {
+    constructor(radius, azimuth, parentRingRadius) {
        this.radius = radius
        this.azimuth = azimuth
-       this.pinholeArea = pinholeArea
-       this.pointX = radius * Math.cos(azimuth); 
-       this.pointY = radius * Math.sin(azimuth);       
+       this.pinholeArea = Math.PI * radius ** 2
+       this.point = new Point(parentRingRadius * Math.cos(azimuth), parentRingRadius * Math.sin(azimuth)); 

```

Comment:
Added back point class, agree isn't totally needed, but is bit more readable in my opinion